lib/jekyll-admin/server/page.rb

```diff
@@ -20,7 +20,9 @@ class Server < Sinatra::Base
         end
 
         write_file(write_path, page_body)
-        updated_page = pages.find { |p| p.path == write_path }
+        updated_page = Jekyll::Page.new(
+          site, site.source, File.dirname(write_path), File.basename(write_path)
```

Comment:
@benbalter The `write_path` comes from either `document_path` or `request_path`. In my opinion this should most certainly be sanitized before it gets here.